src/test/java/hudson/plugins/sonar/SonarBuildWrapperTest.java

```diff
@@ -0,0 +1,164 @@
+    assertThat(wrapper.getInstallation()).isNull();
+  }
+
+  @Test
+  public void testDescriptor() {
+    SonarBuildWrapper wrapper = new SonarBuildWrapper("local");
+    DescriptorImpl desc = wrapper.getDescriptor();
+
+    assertThat(desc.getDisplayName()).isEqualTo("Prepare SonarQube Scanner environment");
+    assertThat(desc.getHelpFile()).isEqualTo("/plugin/sonar/help-buildWrapper.html");
+    assertThat(desc.isApplicable(mock(AbstractProject.class))).isTrue();
+
+    assertThat(desc.getSonarInstallations()).isEmpty();
+  }
+
+  public void testLogging() throws RunnerAbortedException, IOException, InterruptedException {
+    SonarBuildWrapper wrapper = new SonarBuildWrapper("local");
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+    OutputStream os = wrapper.decorateLogger(mock(AbstractBuild.class), bos);
+    IOUtils.write("test password\n", os);
+    assertThat(new String(bos.toByteArray())).isEqualTo("test password\n");
+
+    // with a SQ instance configured (should mask passwords)
+    configureSonar(createTestInstallation());
+    bos = new ByteArrayOutputStream();
+    os = wrapper.decorateLogger(mock(AbstractBuild.class), bos);
+    IOUtils.write("test password\n", os);
+    assertThat(new String(bos.toByteArray())).isEqualTo("test ******\n");
+  }
+
+  @Test
+  public void testEnvironment() {
+    SonarBuildWrapper wrapper = new SonarBuildWrapper("local");
+    SonarInstallation installation = createTestInstallation();
+    PrintStream stream = mock(PrintStream.class);
+    SonarEnvironment env = wrapper.new SonarEnvironment(installation, stream);
+
+    Map<String, String> map = new HashMap<String, String>();
+    map.put("key", "value");
+    env.buildEnvVars(map);
+
+    assertThat(map).containsEntry("SONAR_HOST_URL", "http://localhost:9001");
+    assertThat(map).containsEntry("SONAR_CONFIG_NAME", "local");
+    // variable in the value should be resolved
+    assertThat(map).containsEntry("SONAR_LOGIN", "local");
+    assertThat(map).containsEntry("SONAR_PASSWORD", "password");
+    assertThat(map).containsEntry("SONAR_JDBC_URL", "");
+    assertThat(map).containsEntry("SONAR_JDBC_USERNAME", "");
+    assertThat(map).containsEntry("SONAR_JDBC_PASSWORD", "");
+    assertThat(map).containsEntry("SONAR_ADDITIONAL", "");
+
+    // existing entries still there
+    assertThat(map).containsEntry("key", "value");
+    verify(stream).println(contains("Injecting SonarQube environment variables"));
+  }
+
+  @Test
+  public void testBuild() throws Exception {
+    // set up a free style project with our wrapper that will execute CaptureVarsBuilder
+    SonarBuildWrapper wrapper = new SonarBuildWrapper("local");
+    configureSonar(createTestInstallation());
+    CaptureVarsBuilder b = new CaptureVarsBuilder();
+    FreeStyleProject project = setupFreeStyleProject(b);
+    project.getBuildWrappersList().add(wrapper);
+    Run<?, ?> build = build(project);
+
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS2970)